maze_dataset/tokenization/maze_tokenizer.py

```diff
@@ -437,45 +449,245 @@ def clear_cache(self):
 
 @serializable_dataclass(frozen=True, kw_only=True)
```

Comment:
the `tokenizer_element_tree`, `tokenizer_element_tree`, `name`, and maybe even the vocab should be properties that we serialize (and then check that they match on loading)

Reply:
Adding these as `properties_to_serialize` in `TokenizerElement` would create a lot of duplicate R/W and checking ops. Each nested element would check its entire subtree, which seems unnecessary. Same goes for `name`, which is just a more concise string with the same data as `tokenizer_element_tree`.
I added the `properties_to_serialize` to `MazeTokenizerModular` a couple of commits down, which seems like a better option, but lmk if I'm missing something.

Reply:
my mistake, we only need this on the root `MazeTokenizerModular`